core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -329,11 +331,51 @@ private void tryStart() {
             }
 
             // Wait until the process within the container has become ready for use (e.g. listening on network, log message emitted, etc).
-            waitUntilContainerStarted();
+            try {
+                waitUntilContainerStarted();
+            } catch (Exception e) {
+                logger().debug("Wait strategy threw an exception", e);
+                InspectContainerResponse inspectContainerResponse = null;
+                try {
+                    inspectContainerResponse = dockerClient.inspectContainerCmd(containerId).exec();
+                } catch (NotFoundException notFoundException) {
+                    logger().debug("Container {} not found", containerId, notFoundException);
+                }
+
+                if (inspectContainerResponse == null) {
+                    throw new IllegalStateException("Container is removed");
+                }
+
+                InspectContainerResponse.ContainerState state = inspectContainerResponse.getState();
+                if (Boolean.TRUE.equals(state.getDead())) {
+                    throw new IllegalStateException("Container is dead");
+                }
+
+                if (Boolean.TRUE.equals(state.getOOMKilled())) {
+                    throw new IllegalStateException("Container crashed with out-of-memory");
+                }
+
+                String error = state.getError();
+                if (!StringUtils.isBlank(error)) {
+                    throw new IllegalStateException("Container crashed: " + error);
+                }
+
+                if (!Boolean.TRUE.equals(state.getRunning())) {
+                    throw new IllegalStateException("Container exited with code " + state.getExitCode());
+                }
+
+                throw e;
+            }
 
             logger().info("Container {} started", dockerImageName);
             containerIsStarted(containerInfo);
         } catch (Exception e) {
+            if (e instanceof UndeclaredThrowableException && e.getCause() instanceof Exception) {
```

Comment:
slightly unrelated change but since I was getting a lot of errors while debugging, I could not notice that the logs contained **double** wrapped exceptions

Reply:
Interesting... Good spot.